File: android-app/src/main/java/com/novadownloader/mobile/NotificationActionReceiver.java
package com.novadownloader.mobile;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Build;

public class NotificationActionReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        if (intent == null || intent.getAction() == null) return;

        Intent serviceIntent = new Intent(context, MusicService.class);
        serviceIntent.setAction(intent.getAction());

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            context.startForegroundService(serviceIntent);
        } else {
            context.startService(serviceIntent);
        }
    }
}

File: mobile-rn/android/app/src/main/java/com/novadownloader/rn/NotificationActionReceiver.kt
package com.novadownloader.rn

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

class NotificationActionReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val serviceIntent = Intent(context, MusicService::class.java).apply {
            action = intent.action
        }
        context.startService(serviceIntent)
    }
}
